src/Orleans.Runtime/Hosting/EndpointOptions.cs

```diff
@@ -8,25 +10,97 @@ namespace Orleans.Hosting
     public class EndpointOptions
     { 
         /// <summary>
-        /// The external IP address or host name used for clustering.
-        /// </summary>
-        public string HostNameOrIPAddress { get; set; }
-
-        /// <summary>
-        /// The IP address used for clustering. Will be inferred from <see cref="HostNameOrIPAddress"/> if this is not directly specified.
+        /// The IP address used for clustering.
         /// </summary>
         public IPAddress IPAddress { get; set; }
 
         /// <summary>
         /// The port this silo uses for silo-to-silo communication.
         /// </summary>
-        public int Port { get; set; }
+        public int SiloPort { get; set; }
 
         /// <summary>
         /// The port this silo uses for silo-to-client (gateway) communication. Specify 0 to disable gateway functionality.
         /// </summary>
         public int ProxyPort { get; set; }
 
-        //public bool BindToAny { get; set; }
+        /// <summary>
+        /// The endpoint used to listen for silo to silo communication. 
+        /// If not set will default to <see cref="IPAddress"/> + <see cref="SiloPort"/>
+        /// </summary>
+        public IPEndPoint SiloListeningEndpoint { get; set; }
+
+        /// <summary>
+        /// The endpoint used to listen for silo to silo communication. 
+        /// If not set will default to <see cref="IPAddress"/> + <see cref="ProxyPort"/>
+        /// </summary>
+        public IPEndPoint ProxyListeningEndpoint { get; set; }
+    }
+
+    public static class EndpointOptionsExtensions
+    {
+        public static ISiloHostBuilder ConfigureEndpoints(
+            this ISiloHostBuilder builder,
+            IPAddress ip,
+            int siloPort,
+            int proxyPort)
+        {
+            builder.Configure<EndpointOptions>(options =>
+            {
+                options.IPAddress = ip;
+                options.ProxyPort = proxyPort;
+                options.SiloPort = siloPort;
+            });
+            return builder;
+        }
+
+        public static ISiloHostBuilder ConfigureEndpoints(
+            this ISiloHostBuilder builder, 
+            string addrOrHost, 
```

Comment:
Since we'll have the overload above that takes an `IPAddress`, I think it'll be less confusing if we name this one `hostName`, even if we leave the logic that allows to pass string from of an IP address to it as well.